staging/src/k8s.io/apiserver/pkg/features/kube_features.go

```diff
@@ -294,76 +294,156 @@ func init() {
+	APIServerTracing: {
+		{Version: version.MustParse("1.22"), Default: false, PreRelease: featuregate.Alpha},
+		{Version: version.MustParse("1.27"), Default: true, PreRelease: featuregate.Beta},
+	},
 
-	APIServerTracing: {Default: true, PreRelease: featuregate.Beta},
+	APIServingWithRoutine: {
+		{Version: version.MustParse("1.30"), Default: false, PreRelease: featuregate.Alpha},
+	},
 
-	APIServingWithRoutine: {Default: false, PreRelease: featuregate.Alpha},
+	AuthorizeWithSelectors: {
+		{Version: version.MustParse("1.31"), Default: false, PreRelease: featuregate.Alpha},
+	},
 
-	AuthorizeWithSelectors: {Default: false, PreRelease: featuregate.Alpha},
+	ConcurrentWatchObjectDecode: {
+		{Version: version.MustParse("1.31"), Default: false, PreRelease: featuregate.Beta},
+	},
 
-	ConcurrentWatchObjectDecode: {Default: false, PreRelease: featuregate.Beta},
+	ConsistentListFromCache: {
+		{Version: version.MustParse("1.28"), Default: false, PreRelease: featuregate.Alpha},
+		{Version: version.MustParse("1.31"), Default: true, PreRelease: featuregate.Beta},
+	},
 
-	CoordinatedLeaderElection: {Default: false, PreRelease: featuregate.Alpha},
+	CoordinatedLeaderElection: {
+		{Version: version.MustParse("1.31"), Default: false, PreRelease: featuregate.Alpha},
+	},
 
-	EfficientWatchResumption: {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
+	EfficientWatchResumption: {
+		{Version: version.MustParse("1.20"), Default: false, PreRelease: featuregate.Alpha},
+		{Version: version.MustParse("1.21"), Default: true, PreRelease: featuregate.Beta},
+		{Version: version.MustParse("1.24"), Default: true, PreRelease: featuregate.GA, LockToDefault: true},
+	},
 
-	KMSv1: {Default: false, PreRelease: featuregate.Deprecated},
+	MutatingAdmissionPolicy: {
+		{Version: version.MustParse("1.30"), Default: false, PreRelease: featuregate.Alpha},
+	},
 
-	OpenAPIEnums: {Default: true, PreRelease: featuregate.Beta},
+	OpenAPIEnums: {
+		{Version: version.MustParse("1.23"), Default: false, PreRelease: featuregate.Alpha},
+		{Version: version.MustParse("1.24"), Default: true, PreRelease: featuregate.Beta},
+	},
 
-	RemainingItemCount: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // remove in 1.32
+	RemainingItemCount: {
+		{Version: version.MustParse("1.15"), Default: false, PreRelease: featuregate.Alpha},
+		{Version: version.MustParse("1.16"), Default: true, PreRelease: featuregate.Beta},
+		{Version: version.MustParse("1.29"), Default: true, PreRelease: featuregate.GA, LockToDefault: true},
+	},
 
-	ResilientWatchCacheInitialization: {Default: true, PreRelease: featuregate.Beta},
+	ResilientWatchCacheInitialization: {
+		{Version: version.MustParse("1.31"), Default: true, PreRelease: featuregate.Beta},
+	},
 
-	RetryGenerateName: {Default: true, PreRelease: featuregate.Beta},
+	SeparateCacheWatchRPC: {
+		{Version: version.MustParse("1.30"), Default: true, PreRelease: featuregate.Beta},
```

Comment:
bugfix gate picked back to 1.28 as a mitigation: https://github.com/kubernetes/kubernetes/pull/123694